src/main/java/io/openliberty/tools/common/plugins/config/ServerConfigDocument.java

```diff
@@ -240,17 +243,26 @@ private static void parseNames(Document doc, String expression) throws XPathExpr
 
         for (int i = 0; i < nodeList.getLength(); i++) {
             if (nodeList.item(i).getAttributes().getNamedItem("name") != null) {
-                String nodeValue = nodeList.item(i).getAttributes().getNamedItem("name").getNodeValue();
+                String nameValue = nodeList.item(i).getAttributes().getNamedItem("name").getNodeValue();
+                String locationValue = nodeList.item(i).getAttributes().getNamedItem("location").getNodeValue();
 
                 // add unique values only
-                if (!nodeValue.isEmpty()) {
-                    String resolved = resolveVariables(nodeValue, null);
-                    if (resolved == null) {
-                        if (!names.contains(nodeValue)) {
-                            names.add(nodeValue);
+                if (!nameValue.isEmpty()) {
+                    String resolvedName = resolveVariables(nameValue, null);
+                    String resolvedLocation = resolveVariables(locationValue, null);
+                    if (resolvedName == null) {
+                        if (!names.contains(nameValue)) {
+                            names.add(nameValue);
+                            if (resolvedLocation != null) {
+                                locationsAndNames.put(resolvedLocation, nameValue);
+                            }
+                        }
+                    } else if (!names.contains(resolvedName)) {
+                        names.add(resolvedName);
+                        
+                        if (resolvedLocation != null) {
+                            locationsAndNames.put(resolvedLocation, resolvedName);
```

Comment:
I think perhaps an app could have the same name in multiple locations? I would suggest pulling line 257 and 264 out of the if/else if/else if statement. Put it right after between 266 and 267. Then you will get all locations mapped to their app names.

Reply:
One more point...you need to reinitialize the HashMap in initializeAppsLocation.